### DISCUSSION TITLE
Scoring profiles

### DISCUSSION THREAD

**Discussion by IainD925** [microsoft/AI-Chat-App-Hack#78] [Q&A] [answered]
From the [MS Learn scoring documentation](https://learn.microsoft.com/en-us/azure/search/index-add-scoring-profiles) I understand how to create a scoring profile in JSON but I'm unclear how the data should be structured.

The following text is an example which I'd like to use a scoring profile to adds weight to the 'Tag' field.
How do we get the index to recognise the Tag field?

`Document-Key: 123-abc-999
Document: what-tell-children.pdf
Title: What should we tell children about Santa?
Tag: kids
Text:  The topic of Santa Claus is both magical and complex when it comes to discussing it with children. The approach to take can depend on several factors including the age of the child, the family's cultural or religious beliefs, and personal preferences regarding the celebration of holidays. Here are some general guidelines and ideas on what to tell children about Santa:

* Telling children about Santa can foster a sense of wonder and excitement, encouraging imagination and belief in something greater. You can tell stories of Santa coming from the North Pole, flying with his reindeer, and delivering presents to children around the world.

* Santa can also be a way to teach children about the joy of giving. Explain that Santa embodies the spirit of giving without expecting anything in return. This can be a good opportunity to involve children in choosing and giving gifts to others, showing them the happiness that comes from making others feel loved and appreciated.

* Be Honest at the Right Time
As children grow older, they may start asking more questions or expressing doubts about the existence of Santa. This can be a delicate moment for parents. Some choose to gradually reveal the truth, emphasizing that while Santa might not be a real person, the ideas of kindness, generosity, and magic he represents are very real and valuable. Others might choose to keep the story alive as long as the child believes, allowing them to come to their own conclusions in their own time.

Conclusion
Ultimately, the decision on what to tell children about Santa is a personal one for each family. The key is to create a narrative that feels right for your family, supports your values, and fosters a sense of joy and wonder in the holiday season. Whether Santa is a fun, fictional character or a symbol of the holiday spirit, the stories told can create lasting memories and traditions for children.`

**Comment by pamelafox** [comment]
Have you already added tags as a field in your search index? If not, you'd need to do that and re-index the data, and it looks like you'd need to mark it as filterable=True, per https://learn.microsoft.com/en-us/azure/search/index-add-scoring-profiles#rules-for-using-functions

Or did you do that and it's not working?

**Reply by IainD925** [reply, marked as answer]
Not done, yet.  I didn't understand how the data needed to be structured but found resources that explains the process.

Sharing links, for others:
Search JSON Blobs
[https://learn.microsoft.com/en-us/azure/search/search-semi-structured-data](https://eur03.safelinks.protection.outlook.com/?url=https%3A%2F%2Flearn.microsoft.com%2Fen-us%2Fazure%2Fsearch%2Fsearch-semi-structured-data&data=05%7C02%7Ciain.dobson%40quorum.co.uk%7Cbad8c0b19d1d440b5a3508dc28d1e181%7C43ad7b98e4ac431391355e9a60cca7f0%7C0%7C0%7C638430129094596838%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C0%7C%7C%7C&sdata=xBj9DrXteJiH75LTShimFhg5STyIzxzuOtzRceNua0w%3D&reserved=0)

Tutorial: Index semi-structured data
[https://learn.microsoft.com/en-us/azure/search/search-howto-index-json-blobs](https://eur03.safelinks.protection.outlook.com/?url=https%3A%2F%2Flearn.microsoft.com%2Fen-us%2Fazure%2Fsearch%2Fsearch-howto-index-json-blobs&data=05%7C02%7Ciain.dobson%40quorum.co.uk%7C69091af10e254b2cb38708dc28d1c175%7C43ad7b98e4ac431391355e9a60cca7f0%7C0%7C0%7C638430128560660802%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C0%7C%7C%7C&sdata=4i03tnD8yIrQ0DmnUUEZz%2BWeYT0bz3pE6M1SEuYpJbA%3D&reserved=0)
